src/Nethermind/Nethermind.BeaconNode.Storage/MemoryStore.cs

```diff
@@ -111,31 +116,22 @@ public ValueTask<BeaconState> GetBlockStateAsync(Root blockRoot)
             return new ValueTask<BeaconState?>(state);
         }
 
-        public async IAsyncEnumerable<Root> GetChildKeysAfterSlotAsync(Root parent, Slot slot)
+        public async IAsyncEnumerable<Root> GetChildKeysAsync(Root parent)
         {
             await Task.CompletedTask;
-            IEnumerable<Root> childKeys = _blocks
+            IEnumerable<Root> childKeys = _signedBlocks
                 .Where(kvp =>
-                    kvp.Value.ParentRoot.Equals(parent)
-                    && kvp.Value.Slot > slot)
+                    kvp.Value.Message.ParentRoot.Equals(parent))
                 .Select(kvp => kvp.Key);
             foreach (Root childKey in childKeys)
             {
                 yield return childKey;
             }
         }
 
-        public async IAsyncEnumerable<Root> GetChildKeysAsync(Root parent)
+        public async Task<Root> GetHeadAsync()
```

Comment:
does it not break single responsibility of store?

Reply:
IStore is now more of a facade because the actual implementation of GetHeadAsync() is actually in SimpleLatestMessageDrivenGreedyHeaviestObservedSubtree : IHeadSelectionStrategy

I moved it into the Storage component, along with some of the other parts, because for in-memory it just runs the algorithm, but if it had to access historical storage it might be easier to do that directly in the datastore (e.g. find a parent).

However, maybe from your Eth1 experience you will never actually read from storage that way, but always have everything cached in memory anyway?   (Eth2 may also be different, as it has finalization so there is a limit to possible forks)

The actual code is in the strategy class (and there are a half dozen alternative approaches mentioned in the spec), so implementation is still separate.  Maybe I can clean up the interfaces / names.